### DISCUSSION TITLE
useQuery isLoading state stuck on true

### DISCUSSION THREAD

**Discussion by heyitzkamal** [TanStack/query#1732] [Q&A]
Hi, So i have a useQuery hook which is configured to have a stale time of 5 minutes. Problem is if I manually navigate to the page by entering the url in the search bar, on the first attempt it shows the loading state and then the next time it displays my data. Sometimes not always on the second attempt, it could take 3-5 page refreshes before it displays the data. It is successfully making the network request and fetching the data and i can see the data in the devtools.
 How can i resolve this?

```
export const usePosts = <TData = PostsResponse>(page: number) => {
  const options: UseQueryOptions<PostsResponse, AxiosError> = {
    keepPreviousData: true,
    staleTime       : 1000 * 60 * 5
  };
  return useQuery([ "posts", page ], () => fetchPosts(page), options);
};
```
```
  const { data, isLoading, isPreviousData } = usePosts(page);

  const Loading = <div>Loading...</div>;

  if (isLoading) {
    return Loading;
  }
```

**Comment by heyitzkamal** [comment]
![stale](https://user-images.githubusercontent.com/68304628/106678105-129d5780-65cb-11eb-8378-3a0d47a7f3f9.PNG)

**Comment by VikaramjitSinghGorraya** [comment]
Hi, were you able to find a solution for this? I am also facing the same issue.

**Comment by nawfalhaddi** [comment]
@TkDodo  I noticed this issue is random, but what happens is the observer doesn't update the arguments after receiving the response from the API call

**Comment by eminvergil** [comment]
Hi, I have the same issue. I am using ` "@tanstack/react-query": "^4.0.10"`. Is there a solution to this problem ?

**Reply by TkDodo** [reply]
please show a reproduction, preferably in codesandbox.

**Reply by hungtp1996nd** [reply]
> please show a reproduction, preferably in codesandbox.

Hi @TkDodo, i have same issue with "isLoading" always "true" when i try config with "enabled". 
My csb: https://codesandbox.io/s/react-typescript-forked-lt7ksl

**Reply by TkDodo** [reply]
that's because you set `enabled:false` and that is how it works in v4

https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#the-idle-state-has-been-removed

**Reply by hungtp1996nd** [reply]
with my csb, is rendering "Loading..." wrong, how can i fix it?

**Reply by TkDodo** [reply]
`if (isLoading && isFetching)`

**Reply by hungtp1996nd** [reply]
So sweet, thanks for your support

**Reply by abdulfatahi** [reply]
> `if (isLoading && isFetching)`

Thanks for this

**Reply by ibreaker14** [reply]
You can substitute `if(isLoading && isFetching)` with `isInitialLoading`. It does the same thing under the hood.

https://tanstack.com/query/v4/docs/react/guides/disabling-queries

**Reply by mathura333** [reply]
@TkDodo Is this a bug or expected behavior?
This is happening randomly

"@tanstack/react-query": "^4.16.1"

**Reply by annguyensyncly** [reply]
isLoading && isFetching : All true. fuc  atom

**Comment by bogdanq** [comment]
"@tanstack/react-query": "^4.2.3" - same problem

**Comment by A-Ostrovnyy** [comment]
"@tanstack/react-query": "^4.32.0" - same problem 😢

**Reply by TkDodo** [reply]
It's likely this chromium related issue:

- https://github.com/TanStack/query/issues/5679

which we have worked around in v5. But without you showing some more information (e.g. what the `fetchStatus` of your query is), it's impossible to tell.

**Comment by danielR2001** [comment]
"@tanstack/react-query": "^5.36.0" - same problem 😢 
Some queries are randomly stuck on 'stale' and data = null
even though the query function got a result with data.

**Comment by danielboso** [comment]
I had this problem. I discovered what I think is not a good solution, but it's working. I realized that if I set a state (I called enabled), and was set to false, on useEffect I can change to true. I have forwarded this state into the enabled config of the query

**Reply by stokedout** [reply]
Do you have an example?

**Comment by DBarbero** [comment]
Same problem with `"4.0.10"`.

This is my workaround, hope it helps:

Declare a `isMounted` initialised to false
Use the useEffect hook to set it to true 
Pass it to useQuery enabled option
Now `isLoading` should work as expected

```
const [isMounted, setIsMounted] = useState<boolean>(false)

const { isLoading } = useQuery({
    queryKey: ['get-x],
    queryFn: service.getX,
    enabled: isMounted,
})

useEffect(() => {
    setIsMounted(true)

    return () => {
        setIsMounted(false)
    }
}, [])
```

**Reply by nv-minh** [reply]
nice it work for me thank bro
